tezos_interop/tezos_interop.re

```diff
@@ -172,7 +181,15 @@ module Key_hash = {
     switch (t) {
     | Key.Ed25519(pub_) => Ed25519(Ed25519.Hash.hash_key(pub_))
     };
-
+  let get_hash =
+    fun
+    | Ed25519(h) => h;
```

Comment:
Same as my previous feedback

> I think the keys have to be unwrapped from the respective Tags/Variants and then converted to cstruct right away. Instead of passing down the wrapped key to to_cstruct. This will make it easier in future when we support more curves

You'll see how `get_hash` becomes redundant then. It'll simply be,

```reasonml
  let equal = (a, b) => {
    switch (a, b) {
    | (Ed25519(a), Ed25519(b)) =>
      Int.equal(BLAKE2B_20.compare(a, b), 0)
    };
  };
```

Reply:
updated